pkg/plugin/parsing/parsing.go

```diff
@@ -85,17 +85,15 @@ func ParseData(graphQlResponseData *jsonnode.Object, parsingOption querymodel.Pa
 	fm := framemap.New()
 
 	for _, dataElement := range dataArray {
-		flatData := jsonnode.NewObject()
-		flattenData(dataElement, "", flatData)
-		labels, err := getLabelsFromFlatData(flatData, parsingOption)
+		labels, err := getLabelsFromFlatData(dataElement, parsingOption)
```

Comment:
I know that calling flattenData() will turn the arrays into flat data, which isn't what you want, but calling flattenData is still a requirement because one, it allows the `getLabelsFromFlatData` to function correctly on data that isn't flat, and two it allows the iteration over keys and primitive values, rather than keys and potentially complex JSON objects.